packages/payload/src/fields/config/types.ts

```diff
@@ -267,7 +267,7 @@ type Admin = {
   position?: 'sidebar'
   readOnly?: boolean
   style?: CSSProperties
-  width?: string
+  width?: `${string}${'%' | 'px'}`
 }
```

Comment:
Yeah good point you're 100% right about the units. 

How about this?

```ts
width?: `${number}${'%' | 'ch' | 'em' | 'pc' | 'pt' | 'px' | 'rem' | 'vh' | 'vmax' | 'vmin' | 'vw'}`
```

Technically there is also `fr` but that wouldn't make sense because it's only for `display: grid`

https://github.com/payloadcms/payload/pull/7940/commits/ef6fe9ca9a55b54d1d634b3421f9fd80881ae0d8#diff-4eb3539a66d16f5cd16f3833db718072b1950c2d41d5a3fe259877d3153205cbR270

Another idea I had was to also allow for `clamp()` and `calc()` at a type level like this: https://www.typescriptlang.org/play/?#code/C4TwDgpgBAqgdgS2FAvAAwCQG84FcC2ARhAE4C+2A5AKSVQA+UlAxgBZ2OUT4dNjO9KYYILAAPQSW6CAbuwZMZ+AIYSFlJQjiyA7pTJoAUKEhQAwgBtl+MKihpmVmwAps8JGQA0UN4mBefLHd-AEojE2gzZQtmOwdo5lcsAGdgEi0AczIww2NwaAAlCGTcC2QUcydbRmCFKJjDZgB7OFSoYGLgAC4oIpKyuxYq5wBGAAYZHW9xyemJnRDKIA

but maybe that's overkill?

Reply:
i personally feel like it doesn't make too much sense to restrict the type to be just `%` because if the developer passes for example `400px` it will actually set the `width` to that value at runtime

Reply:
hm, on the one hand, I get the feeling that whoever uses an absolute unit like `px` is not thinking about making the design responsive, also, the tests are only made for % and I'm not sure if there can't be an edge case with another unit. On the other hand, I like to give freedom to the users.

What do you guys think @AlessioGr @jacobsfletch

Reply:
Here's probably the reason why this CSS TS properties aren't typed like this:   https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1179#issuecomment-952642724, just to think.

Reply:
Interesting. I can see why they blocked that PR but on the other hand I can't think of any reason why a developer would want to dynamically set the width like `6 + 'px'` in a static payload config file (or why we would want to support that). 

That PR is trying to type the width of DOM elements (massive ecosystem-wide change) whereas this is just typing the payload config object (which is something payload has complete control over)

I think the current suggestion is good because it alerts the developer in the right direction if they make a typo `width: 3o%` and at the same time it allows them flexibility if they really want to use another unit of measurement like `px` (i.e. not recommended but maybe they have a usecase for it and technically it works)

Reply:
This should be typed as `width?: CSSProperties['width']` so that Payload is not the source of truth when it comes to CSS types.

Reply:
addressed here: https://github.com/payloadcms/payload/pull/7940/commits/37e181a38d3d2975da9bfd8ff1f68c00de16755e#diff-4eb3539a66d16f5cd16f3833db718072b1950c2d41d5a3fe259877d3153205cbR270